tests/unittests/test_backoff.py

```diff
@@ -808,3 +812,416 @@ def test_no_socket_timeout_error_occurred__subscriber(self, mocked_write_records
+        obj = lists.ListDataAccessObject({'start_date': '2021-01-01T00:00:00Z'}, {}, None, {})
+        try:
+            # call sync
+            obj.sync_data()
+        except URLError:
+            pass
+        # verify the code backed off and requested for 5 times
+        self.assertEquals(mocked_get.call_count, 5)
+
+    @mock.patch("FuelSDK.rest.ET_GetSupport.get")
+    @mock.patch("tap_exacttarget.dao.DataAccessObject.write_records_with_transform")
+    def test_no_timeout_error_occurred__list(self, mocked_write_records, mocked_get, mocked_sleep):
+        # mock 'get' and return the dummy data
+        mocked_get.side_effect = [get_response(True, [{
+            "CategoryID": 12345,
+            "ContentCheckStatus": "Not Checked",
+            "CreatedDate": "2021-01-01T00:00:00Z",
+            "EmailType": "Normal"
+        }])]
+        # make the object of 'ListDataAccessObject'
+        obj = lists.ListDataAccessObject({'start_date': '2021-01-01T00:00:00Z'}, {}, None, {})
+        # call sync
+        obj.sync_data()
+        # verify if 'tap_exacttarget.dao.DataAccessObject.write_records_with_transform' was called
+        # once as there is only one record
+        self.assertEquals(mocked_write_records.call_count, 1)
+
+    @mock.patch("FuelSDK.rest.ET_GetSupport.get")
+    def test_timeout_error_occurred__sends(self, mocked_get, mocked_sleep):
+        # mock 'get' and raise error
+        mocked_get.side_effect = URLError('_ssl.c:1059: The handshake operation timed out')
+        # make the object of 'SendDataAccessObject'
+        obj = sends.SendDataAccessObject({'start_date': '2021-01-01T00:00:00Z'}, {}, None, {})
+        try:
+            # call sync
+            obj.sync_data()
+        except URLError:
+            pass
+        # verify the code backed off and requested for 5 times
+        self.assertEquals(mocked_get.call_count, 5)
+
+    @mock.patch("FuelSDK.rest.ET_GetSupport.get")
+    @mock.patch("tap_exacttarget.dao.DataAccessObject.write_records_with_transform")
+    def test_no_timeout_error_occurred__sends(self, mocked_write_records, mocked_get, mocked_sleep):
+        # mock 'get' and return the dummy data
+        mocked_get.side_effect = [get_response(True, [{
+            "CategoryID": 12345,
+            "ContentCheckStatus": "Not Checked",
+            "CreatedDate": "2021-01-01T00:00:00Z",
+            "EmailType": "Normal"
+        }])]
+        # make the object of 'SendDataAccessObject'
+        obj = sends.SendDataAccessObject({'start_date': '2021-01-01T00:00:00Z'}, {}, None, {})
+        # call sync
+        obj.sync_data()
+        # verify if 'tap_exacttarget.dao.DataAccessObject.write_records_with_transform' was called
+        # once as there is only one record
+        self.assertEquals(mocked_write_records.call_count, 1)
+
+    @mock.patch("FuelSDK.rest.ET_GetSupport.get")
+    def test_timeout_error_occurred__subscriber(self, mocked_get, mocked_sleep):
```

Comment:
please add function comments for all tests

Reply:
Added